rfc/rfc023-x509credential.md

```diff
@@ -0,0 +1,352 @@
+    phishing attacks or man-in-the-middle (MitM) attacks).
+  - Sensitive data (e.g., credentials, financial data) exchanged with fraudulent sites could be intercepted.
+
+### 6.2 **Revocation Checks Not Performed**
+
+- If you fail to check the status of certificates for revocation using CRL (Certificate Revocation List), certificates
+  that have been compromised or expired could still be considered valid.
+- **Consequences**:
+  - Attackers could use stolen or revoked certificates to bypass authentication or encryption.
+  - Systems may continue to trust certificates issued to malicious actors.
+
+### 6.3 **Expired Certificates**
+
+- If credential expiry is ignored, certificates whose validity period has elapsed could still be used and trusted.
+- **Consequences**:
+  - Attackers may exploit outdated certificates to perform replay attacks where previously valid credentials are reused.
+  - Trust in infrastructure degrades because expired certificates no longer reflect proper certificate holder
+    responsibility/accountability.
+
+### 6.4 **Weak Keys or Algorithms**
+
+- If weak cryptographic algorithms (e.g., MD5, SHA-1) or small key sizes (e.g., <2048-bit RSA) are used, the
+  certificates or their signatures could be cracked by modern computational power.
+- **Consequences**:
+  - An attacker could forge or spoof certificates.
+  - Sensitive data could be decrypted easily, exposing confidential information such as passwords, personal data, etc.
+
+### 6.5 **Improper Credential Subject Validation**
+
+- If the `credentialSubject` field in frameworks like `X509Credential` is not properly validated, it may allow
+  fields not aligned with the X.509 certificate to be added or accepted.
+- **Consequences**:
+  - Attackers could inject unauthorized or false data (e.g., incorrect organization name or purpose), tricking verifiers
+    by impersonating trusted entities.
+  - Loss of trust in the system due to inconsistencies between certificates and credentials.
+
+### 6.6 **Forged Proofs or Tampered Credentials**
+
+- Failure to verify cryptographic proofs tied to certificates could allow credentials or data to be tampered with.
+- **Consequences**:
+  - Credentials could be modified to grant unauthorized access.
+  - The integrity of systems relying on these credentials could be compromised.
+
+### 6.7 **Missing Root CA Verification**
+
+- If the source of trust (Root CA) is not explicitly verified and trusted, attackers could use certificates issued by
+  unapproved CAs.For instance, in case of UZI certificates the `ca-fingerprint` must match the hash of either the Root
+  CA or one of the Intermediate CAs as published by the [UZI register](https://www.zorgcsp.nl/ca-certificaten).
+- **Consequences**:
+  - Fake certificates signed by rogue or unvalidated CAs could be accepted as valid.
+  - Attackers gain the ability to impersonate legitimate entities in scenarios such as encrypted communication or
+    identity verification.
```

Comment:
I think this is a bad example. UZI server certificates must be signed by the intermediate issuing the leaf or you would pin the entire PKIo chain and not just the UZI server certs.

It is also not clear to me what this point is about. It starts on root CA ( which suggests chain) validation, but it the example is on certificate pinning